### PR TITLE
[DO-557] handle trailing slashes

### PR DESCRIPTION
**Pull request by StanG4313**

### CODE REVIEW

File: utils/wTrailingSlash.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { withTrailingSlash } from 'ufo'
+
+function wTrailingSlash(str: string | undefined): string | undefined {
+  if (typeof str === 'string') {
+    const result = withTrailingSlash(str)
+    return result
+  } else {
+    return undefined
+  }
+}
```
[the remark this answers is not in the document]
Comment (StanG4313) [reply utils/wTrailingSlash.ts:26]: Without undefined option I receive type error, that's why I decided to leave this option

[the remark this answers is not in the document]
Comment (d0rich) [reply utils/wTrailingSlash.ts:26]: You still can handle this case outside of function. It seems in some cases input value can be undefined.

[the remark this answers is not in the document]
Comment (StanG4313) [reply utils/wTrailingSlash.ts:26]: OK, will try to find a decent work-around

File: utils/wTrailingSlash.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { withTrailingSlash } from 'ufo'
+
+function wTrailingSlash(str: string | undefined): string | undefined {
+  if (typeof str === 'string') {
+    const result = withTrailingSlash(str)
```
[the remark this answers is not in the document]
Comment (StanG4313) [reply utils/wTrailingSlash.ts]: Renamed on next commit

File: utils/wTrailingSlash.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (StanG4313) [reply utils/wTrailingSlash.ts]: Now fixed

File: components/ep/layout/SurroundDocCard.vue [hunk 3]
```diff
@@ -31,7 +31,9 @@ defineProps<{
 
 <template>
   <NuxtLink
-    :to="wTrailingSlash(doc._path)"
+    :to="
+      typeof doc._path === 'string' ? withTrailingSlash(doc._path) : doc._path
+    "
```
Comment (d0rich) [inline components/ep/layout/SurroundDocCard.vue:36]: There can be a shorten expression

> If doc._path is `undefined` provide `''` instead.

```suggestion
    :to="withTrailingSlash(doc._path ?? '')"
```

**Review by d0rich — CHANGES_REQUESTED** [review]

**Review by d0rich — APPROVED** [review]

**Pull request merged**